File: backend/src/main/java/com/example/backend/domain/voter/exception/VoterException.java
package com.example.backend.domain.voter.exception;

import org.springframework.http.HttpStatus;

public class VoterException extends RuntimeException {
    private final VoterErrorCode voterErrorCode;

    public VoterException(VoterErrorCode voterErrorCode) {
        super(voterErrorCode.getMessage());
        this.voterErrorCode = voterErrorCode;
    }

    public HttpStatus getStatus() {
        return voterErrorCode.getHttpStatus();
    }

    public String getCode() {
        return voterErrorCode.getCode();
    }
}

File: backend/src/main/java/com/example/backend/domain/voter/exception/VoterException.kt
package com.example.backend.domain.voter.exception

import org.springframework.http.HttpStatus

class VoterException(
    val voterErrorCode: VoterErrorCode
) : RuntimeException(voterErrorCode.message) {

    fun getStatus(): HttpStatus = voterErrorCode.httpStatus

    fun getCode(): String = voterErrorCode.code
}
